var/spack/repos/builtin/packages/py-numba/package.py

```diff
@@ -21,29 +23,30 @@ class PyNumba(PythonPackage):
     version("0.48.0", sha256="9d21bc77e67006b5723052840c88cc59248e079a907cc68f1a1a264e1eaba017")
     version("0.40.1", sha256="52d046c13bcf0de79dbfb936874b7228f141b9b8e3447cc35855e9ad3e12aa33")
 
-    depends_on("python@3.7:3.10", type=("build", "run"), when="@0.55.0:")
-    depends_on("python@3.7:3.9", type=("build", "run"), when="@0.54")
-    depends_on("python@3.6:3.9", type=("build", "run"), when="@0.53")
-    depends_on("python@3.6:3.8", type=("build", "run"), when="@0.52")
+    depends_on("python@3.7:3.10", when="@0.55.0:", type=("build", "run"))
+    depends_on("python@3.7:3.9", when="@0.54", type=("build", "run"))
+    depends_on("python@3.6:3.9", when="@0.53", type=("build", "run"))
+    depends_on("python@3.6:3.8", when="@0.52", type=("build", "run"))
     # set upper bound for python the same as newer release
-    depends_on("python@3.6:3.8", type=("build", "run"), when="@0.48:0.51")
-    depends_on("python@3.3:3.7", type=("build", "run"), when="@0.40.1:0.47")
-    depends_on("py-numpy@1.18:1.22", type=("build", "run"), when="@0.55.2:")
-    depends_on("py-numpy@1.18:1.21", type=("build", "run"), when="@0.55.0:0.55.1")
-    depends_on("py-numpy@1.17:1.20", type=("build", "run"), when="@0.54")
+    depends_on("python@3.6:3.8", when="@0.48:0.51", type=("build", "run"))
+    depends_on("python@3.3:3.7", when="@0.40.1:0.47", type=("build", "run"))
+    depends_on("py-numpy@1.18:1.23", when="@0.56.1:", type=("build", "run"))
```

Comment:
It seems like this line is not doing anything, since the following like has a stricter dependency range for a wider set of versions. Should py-numpy@1.23 be allowed with py-numba@0.56.4?

Reply:
Ah yes, the line after this should be changed to `when="@0.55.2:0.56.0"`. Can you submit a PR to fix this?